### PR TITLE
Rewrite using chakra-ui

### PR DESCRIPTION
**Pull request by mohitkyadav**
- Added light/dark mode
- Responsive
- Added icon buttons
- Add section for actions modal
- Add launching date section with email

@MitchellMarkGeorge You also need to enable deployments for react on netlify.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -5,6 +5,8 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <title>Tab Butler - a browser tab manager at your service</title>
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
```
Comment (MitchellMarkGeorge) [inline index.html]: Hey why are these needed?

File: public/manifest.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    {
+      "src": "logo512.png",
+      "type": "image/png",
+      "sizes": "512x512"
+    }
+  ],
+  "start_url": ".",
+  "display": "standalone",
+  "theme_color": "#000000",
+  "background_color": "#ffffff"
```
Comment (MitchellMarkGeorge) [inline public/manifest.json]: One thing we are going to have to do before beta is add in the favicon. For now it should be fine.

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+} from '@chakra-ui/react';
+
+import { HomePage } from './Pages';
+
+function App() {
+  return (
+    <ChakraProvider theme={extendTheme({
+      fonts: {
+        body: `'Open Sans', sans-serif`,
+      },
```
Comment (MitchellMarkGeorge) [inline src/App.js]: It would be better if the font stayed consistent with what is already used in the Tab Butler extension itself.

File: src/Pages/HomePage.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    <Box textAlign="center" fontSize="xl">
+      <VStack>
+        <NavBar />
+        <VStack spacing={8}>
+          <Stack
+            spacing={8}
+            mt={20}
+            pl="10"
+            pr="10"
+            direction={{ base: "column", lg: "row"}}
```
Comment (MitchellMarkGeorge) [inline src/Pages/HomePage.js]: As a fan of Chakra UI I understand what these props are but in the case of other developers do you mind using their full names?

File: src/Pages/HomePage.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+                The last browser extension you will need to navigate in your browser.
+              </Text>
+            </VStack>
+            <Image
+              src={`/static/img/search-modal-${theme}.png`}
+              sx={{
+                height: "min(29.4vw + 70.78px,447px)",
+                width: "auto",
+                objectFit: "contain",
+              }}
```
Comment (MitchellMarkGeorge) [inline src/Pages/HomePage.js]: Please what is this meant to do exactly?

File: src/components/NavBar/ColorModeSwitcher.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    <IconButton
+      aria-label={`Switch to ${text} mode`}
+      variant="ghost"
+      opacity={0.75}
+      onClick={toggleColorMode}
+      icon={<SwitchIcon />}
+      {...props}
+    />
+  );
+};
```
Comment (MitchellMarkGeorge) [inline src/components/NavBar/ColorModeSwitcher.js]: I like this!!! 👍

File: src/components/NavBar/NavBar.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+export const NavBar = () => {
+  return (
+    <HStack
+      maxW={1080}
+      justifyContent="space-between"
+      pr={["5px", "30px"]}
+      pl={["5px", "30px"]}
+      pt={["10px", "30px"]}
+      w="100%"
```
Comment (MitchellMarkGeorge) [inline src/components/NavBar/NavBar.js]: Same as above

**Review by MitchellMarkGeorge — CHANGES_REQUESTED** [review]
Overall great work @mohitkyadav just made. a few comments here and there. I will update Netlify to allow React in a bit.

**Comment by MitchellMarkGeorge** [comment]
@mohitkyadav Since this has become a full React app, we should consider using TypeScript in the project down the line.

File: index.html (hunk 1 above)
Comment (mohitkyadav) [reply index.html:9]: These comes with the font.
Screenshot 2022-07-11 at 8 01 19 AM

File: public/manifest.json (hunk 2 above)
Comment (mohitkyadav) [reply public/manifest.json]: Yeah for that, I'll just use the current one for now.

File: src/App.js (hunk 3 above)
Comment (mohitkyadav) [reply src/App.js]: Is it Roboto in the extension?

File: src/Pages/HomePage.js (hunk 5 above)
Comment (mohitkyadav) [reply src/Pages/HomePage.js]: This makes the responsive, so it wouldn't stretch, tear, overflow and crop when you resize the page.

**Comment by mohitkyadav** [comment]
> @mohitkyadav Since this has become a full React app, we should consider using TypeScript in the project down the line.

Done

**Comment by mohitkyadav** [comment]
@MitchellMarkGeorge what does the join beta button do?

File: index.html (hunk 1 above)
Comment (mohitkyadav) [reply index.html]: Removed the old index.html.

File: src/App.js (hunk 3 above)
Comment (MitchellMarkGeorge) [reply src/App.js]: @mohitkyadav No for the font we use the system font (Segoe UI for Windows and SF Pro for Mac and a few others). We do that using this font-family: `-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto,
              Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;`

Here is a guide that explains it better: https://www.digitalocean.com/community/tutorials/css-system-font-stack

File: src/Pages/HomePage.js (hunk 5 above)
Comment (MitchellMarkGeorge) [reply src/Pages/HomePage.js]: Ah I see. That's actually a problem the current site has. Thanks 👍

**Comment by MitchellMarkGeorge** [comment]
> @MitchellMarkGeorge what does the join beta button do?

@mohitkyadav As of right now it should be "inactive" and just show that the beta launches on August 1st. This is what it looks like currently on the website:
Screen Shot 2022-07-11 at 9 35 49 AM

**Comment by MitchellMarkGeorge** [comment]
> * Added light/dark mode
> * Responsive
> * Added icon buttons
> * Add section for actions modal
> 
> TODO:
> 
> * Add launching date section with email
> 
> @MitchellMarkGeorge You also need to enable deployments for react on netlify.

@mohitkyadav In terms of the launch date with email what do you have in mind for that functionality?

**Comment by mohitkyadav** [comment]
@MitchellMarkGeorge For now just the email section is missing. Maybe I can add that in a different PR.

> In terms of the launch date with email what do you have in mind for that functionality?

I think the disabled button with launch date is enough and I'll add a section similar to current one, just with a button "Email me" and some text.

Do you already have a discord server for this?

**Comment by MitchellMarkGeorge** [comment]
> m



> @MitchellMarkGeorge For now just the email section is missing. Maybe I can add that in a different PR.
> 
> > In terms of the launch date with email what do you have in mind for that functionality?
> 
> I think the disabled button with launch date is enough and I'll add a section similar to current one, just with a button "Email me" and some text.
> 
> Do you already have a discord server for this?

Yea that sounds good to me 👍 

Right now I don't have Discord server for this and I'm not that good at running them. If you would like you can go ahead and make one. I can definitely see how having one could be beneficial to users.

**Comment by mohitkyadav** [comment]
Okay then, Also added email section, if there are no further changes I think this PR can be merged. And here's the discord server link. https://discord.gg/vf8mfTTNN3

@MitchellMarkGeorge

**Comment by MitchellMarkGeorge** [comment]
Great work man! I'll merge it in, check it out, then I'll deploy it.

**Comment by MitchellMarkGeorge** [comment]
@mohitkyadav Thanks for making the discord as well man! 🎖️
